skfp/fingerprints/map.py

```diff
@@ -139,135 +135,115 @@ def __init__(
         )
         self.fp_size = fp_size
         self.radius = radius
-        self.variant = variant
+        self.include_duplicated_shingles = include_duplicated_shingles
+        self._encoder: MHFPEncoder = MHFPEncoder(self.fp_size, seed=random_state)
```

Comment:
This probably should be in `._calculate_fingerprint`. Note that if you e.g. do `.set_params()` with current code, the `MHFPEncoder` won't be updated, so this breaks scikit-learn compatibility.

Reply:
Oke